record-and-playback/core/lib/recordandplayback/edl/audio.rb

```diff
@@ -66,124 +68,166 @@ def self.render(edl, output_basename)
+                filter_lines << line
+                track_labels << "[#{track_label}]"
+              end
             end
-
-            # Skip this input and generate silence if the seekpoint is past the end of the audio, which can happen
-            # if events are slightly misaligned and you get unlucky with a start/stop or chapter break.
-            if audio[:timestamp] < (audioinfo[audio[:filename]][:duration] * speed)
-              input_index = ffmpeg_inputs.length
-              ffmpeg_inputs << {
-                filename: audio[:filename],
-                seek: seek
-              }
-              ffmpeg_filter << "[#{input_index}]#{FFMPEG_AFORMAT},apad"
+      
+            # Mix audios if more than one
+            if track_labels.size > 1
+              # e.g.: [t0_0][t0_1]amix=inputs=2[seg0]
+              line = track_labels.join
+              line << "amix=inputs=#{track_labels.size}:normalize=0[#{seg_label}];"
+              filter_lines << line
             else
-              ffmpeg_filter << "#{FFMPEG_AEVALSRC},#{FFMPEG_AFORMAT}"
+              # Only one track => rename it to [segX] (via anull or direct rename)
+              line = "#{track_labels.first}anull[#{seg_label}];"
+              filter_lines << line
             end
-
-            ffmpeg_filter << ',asetpts=N'
-
-            ffmpeg_filter << ",atempo=#{speed},atrim=start=#{ms_to_s(audio[:timestamp])}" if speed != 1
           else
             BigBlueButton.logger.info "  Generating silence"
-
-            ffmpeg_filter << "#{FFMPEG_AEVALSRC},#{FFMPEG_AFORMAT}"
-          end
-
-          if i > 0
-            ffmpeg_filter << "[a_edl#{i}];\n"
-            ffmpeg_filter << "[a_edl#{i}_prev][a_edl#{i}]concat=n=2:a=1:v=0"
+            line = "#{FFMPEG_AEVALSRC},#{FFMPEG_AFORMAT},asetpts=N[#{seg_label}];"
+            filter_lines << line
           end
-
-          ffmpeg_filter << ",atrim=end=#{ms_to_s(entry[:next_timestamp])}"
+      
+          # Now trim this segment to seg_duration
+          trim_line = "[#{seg_label}]atrim=end=#{ms_to_s(seg_duration)}[#{seg_label}];"
+          filter_lines << trim_line
+      
+          # Collect the segment’s final label for the eventual concat
+          segment_labels << "[#{seg_label}]"
         end
-
+      
+        # Build the final concat line if we have at least one segment
+        if segment_labels.any?
+          # e.g.: [seg0][seg1][seg2]concat=n=3:a=1:v=0
+          line = segment_labels.join
+          line << "concat=n=#{segment_labels.size}:a=1:v=0"
+          filter_lines << line
+        end
+      
+        #-------------------------------------
+        # Build the ffmpeg command
+        #-------------------------------------
         ffmpeg_cmd = [*FFMPEG]
         ffmpeg_inputs.each do |input|
-          ffmpeg_cmd += ['-ss', ms_to_s(input[:seek])]
+          # Seek
+          if input[:seek].positive?
+            ffmpeg_cmd += ['-ss', ms_to_s(input[:seek])]
+          end
```

Comment:
I recommend unconditionally adding the seek option. Seeking to 0 slightly changes ffmpeg's timestamp behaviour compared to not seeking, in a way that I believe should make it slightly more reliable at aligning audio files.